assets/src/edit-story/components/inspector/prepublish/components/checklistTab.js

```diff
@@ -243,81 +212,81 @@ const ChecklistTab = ({
+  }
+
+  return (
     <>
-      <SimplePanel
-        isToggleDisabled={isHighPriorityDisabled}
-        name="checklist"
-        hasBadge
-        title={
-          <>
-            <PanelTitle isDisabled={isHighPriorityDisabled}>
-              {TEXT.HIGH_PRIORITY_TITLE}
-            </PanelTitle>
-            {!isHighPriorityDisabled && (
+      {isHighPriorityVisible && (
+        <SimplePanel
+          name="checklist"
+          hasBadge
+          title={
+            <>
+              <PanelTitle>{TEXT.HIGH_PRIORITY_TITLE}</PanelTitle>
               <NumberBadge number={highPriorityLength} />
-            )}
-          </>
-        }
-        ariaLabel={TEXT.HIGH_PRIORITY_TITLE}
-      >
-        {!isHighPriorityDisabled && (
-          <Entries
-            items={highPriority}
-            pageItems={pages.highPriority}
-            renderRow={renderRow}
-            renderPageGroupedRow={renderPageGroupedRow}
-            disabled={isHighPriorityDisabled}
-          />
-        )}
-      </SimplePanel>
-      <SimplePanel
-        isToggleDisabled={isRecommendedDisabled}
-        name="checklist"
-        hasBadge
-        title={
-          <>
-            <PanelTitle isRecommended isDisabled={isRecommendedDisabled}>
-              {TEXT.RECOMMENDED_TITLE}
-            </PanelTitle>
-            {!isRecommendedDisabled && (
-              <NumberBadge isRecommended number={recommendedLength} />
-            )}
-          </>
-        }
-        ariaLabel={TEXT.RECOMMENDED_TITLE}
-      >
-        {canOptimizeVideo && (
-          <AutoVideoOptimization
-            areVideosAutoOptimized={areVideosAutoOptimized}
-            onAutoOptimizeVideoClick={onAutoVideoOptimizationClick}
-          />
-        )}
-        {!isRecommendedDisabled && (
-          <Entries
-            items={recommended}
-            pageItems={pages.recommended}
-            renderRow={renderRow}
-            renderPageGroupedRow={renderPageGroupedRow}
-            disabled={isRecommendedDisabled}
-          />
-        )}
-      </SimplePanel>
+            </>
+          }
+          ariaLabel={TEXT.HIGH_PRIORITY_TITLE}
+        >
+          {highPriority.map(renderRow)}
+          {Object.entries(pages.highPriority || {}).map(renderPageGroupedRow)}
+        </SimplePanel>
+      )}
+      {isRecommendedVisible && (
+        <SimplePanel
+          name="checklist"
+          hasBadge
+          title={
+            <>
+              <PanelTitle isRecommended>{TEXT.RECOMMENDED_TITLE}</PanelTitle>
+              {recommendedLength > 0 && (
+                <NumberBadge isRecommended number={recommendedLength} />
+              )}
+            </>
+          }
+          ariaLabel={TEXT.RECOMMENDED_TITLE}
+        >
+          {canOptimizeVideo && (
+            <AutoVideoOptimization
+              areVideosAutoOptimized={areVideosAutoOptimized}
+              onAutoOptimizeVideoClick={onAutoVideoOptimizationClick}
+            />
+          )}
```

Comment:
💭 I think the toggle should be hidden if there are no videos in the story. This may have existed before this refactor too, sorry!
![image](https://user-images.githubusercontent.com/22185279/117198906-cff44280-ada6-11eb-9e97-98ba627ec4d0.png)

Reply:
i was thinking about this a little, i think it makes sense to leave it visible if it's not been checked yet as it will prevent asks later on.